controllers/spacerequest/spacerequest_controller_test.go

```diff
@@ -25,43 +27,907 @@ import (
 )
 
 func TestCreateSpaceRequest(t *testing.T) {
-
 	// given
 	logf.SetLogger(zap.New(zap.UseDevMode(true)))
 	err := apis.AddToScheme(scheme.Scheme)
 	require.NoError(t, err)
-	basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+	appstudioTier := tiertest.AppStudioTier(t, tiertest.AppStudioTemplates)
 	t.Run("success", func(t *testing.T) {
 		// given
-		sr := spacerequesttest.NewSpaceRequest("jane", "jane-tenant", spacerequesttest.WithTierName("basic"), spacerequesttest.WithTargetClusterRoles([]string{commoncluster.RoleLabel(commoncluster.Tenant)}))
-		member1 := NewMemberClusterWithTenantRole(t, "member-1", corev1.ConditionTrue)
-		member2 := NewMemberClusterWithTenantRole(t, "member-2", corev1.ConditionTrue)
-		hostClient := test.NewFakeClient(t, sr, basicTier)
-		ctrl := newReconciler(hostClient, member1, member2)
-
-		// when
-		res, err := ctrl.Reconcile(context.TODO(), requestFor(sr))
-
-		// then
-		require.NoError(t, err)
-		assert.False(t, res.Requeue)
+		srNamespace := newNamespace("jane")
+		srClusterRoles := []string{commoncluster.RoleLabel(commoncluster.Tenant)}
+		sr := spacerequesttest.NewSpaceRequest("jane", srNamespace.GetName(),
+			spacerequesttest.WithTierName("appstudio"),
+			spacerequesttest.WithTargetClusterRoles(srClusterRoles))
+		parentSpace := spacetest.NewSpace("jane")
+		member1 := NewMemberClusterWithClient(test.NewFakeClient(t, sr, srNamespace), "member-1", corev1.ConditionTrue)
+		hostClient := test.NewFakeClient(t, appstudioTier, parentSpace)
+		ctrl := newReconciler(hostClient, member1)
+
+		t.Run("space doesn't exists it should be created", func(t *testing.T) {
+			// when
+			_, err := ctrl.Reconcile(context.TODO(), requestFor(sr))
+
+			// then
+			require.NoError(t, err)
+			// spacerequest exists with expected cluster roles and finalizer
+			spacerequesttest.AssertThatSpaceRequest(t, srNamespace.Name, sr.GetName(), member1.Client).
+				Exists().
+				HasSpecTargetClusterRoles(srClusterRoles).
+				HasSpecTierName("appstudio").
+				HasFinalizer()
```

Comment:
done please see https://github.com/codeready-toolchain/host-operator/pull/770/commits/443103fe9267c36c433c15299215e5f7d19ef366

I've created a `HasSubSpace` utility function for that, since I don't know the name of the subSpace that will be created.

Reply:
I've renamed `HasSubSpace` to `ContainsSubSpace` in [b43ad74](https://github.com/codeready-toolchain/host-operator/pull/770/commits/b43ad74be73ae098a9d12a2dabf06f367229f6a3) 

sorry for the confusion.